### PR TITLE
Drivetrain

### PR DESCRIPTION
**Pull request by UnknownSilicon**

### CODE REVIEW

File: src/org/robockets/steamworks/OI.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 
+import edu.wpi.first.wpilibj.Joystick;
+
 /**
  * This class is the glue that binds the controls on the physical operator
  * interface to the commands and command groups that allow control of the robot.
  */
 public class OI {
-
+    public static Joystick joystick = new Joystick(0);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/OI.java]: Port number should be in its own constant.

File: src/org/robockets/steamworks/RobotMap.java [hunk 2]
```diff
@@ -8,4 +11,10 @@
  */
 public class RobotMap {
 
+    public static TalonSRX frontLeft = new TalonSRX(0);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/RobotMap.java]: Port number should be in its own constant.

File: src/org/robockets/steamworks/RobotMap.java [hunk 3]
```diff
@@ -8,4 +11,10 @@
...
+    public static TalonSRX frontRight = new TalonSRX(1);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/RobotMap.java]: Port number should be in its own constant.

File: src/org/robockets/steamworks/RobotMap.java [hunk 4]
```diff
@@ -8,4 +11,10 @@
...
+    public static TalonSRX backLeft = new TalonSRX(2);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/RobotMap.java]: Port number should be in its own constant.

File: src/org/robockets/steamworks/RobotMap.java [hunk 5]
```diff
@@ -8,4 +11,10 @@
...
+    public static TalonSRX backRight = new TalonSRX(3);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/RobotMap.java]: Port numbers should be in their own constants.

File: src/org/robockets/steamworks/commands/Drive.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        this.speed = speed;
+    }
+
+    protected void initialize() {
+
+    }
+
+    protected void execute() {
+        if (direction == YAxisRelativeDirection.FORWARD) {
+            Robot.drivetrain.driveArcade(0.5*speed, offset*0.5*speed);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/Drive.java]: What is 0.5?

File: src/org/robockets/steamworks/commands/GottaGoFast.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        this.speed = speed;
+    }
+
+    protected void initialize() {
+        translate = 0;
+        rotate = 0;
+    }
+
+    protected void execute() {
+        translate = OI.joystick.getRawAxis(1);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/GottaGoFast.java]: Button number should be in its own constant.

File: src/org/robockets/steamworks/commands/GottaGoFast.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        rotate = OI.joystick.getRawAxis(4);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/GottaGoFast.java]: Button number should be in its own constant.

**Review by simon-andrews — CHANGES_REQUESTED** [review]
More enums and constants please.

File: src/org/robockets/steamworks/Robot.java [hunk 9]
```diff
@@ -29,8 +34,11 @@
 	@Override
 	public void robotInit() {
 		oi = new OI();
+		drivetrain = new Drivetrain();
 		// chooser.addObject("My Auto", new MyAutoCommand());
 		SmartDashboard.putData("Auto mode", chooser);
+
+		drive = new GottaGoFast();
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/Robot.java]: Doesn't match up with constructor.

File: src/org/robockets/steamworks/commands/GottaGoFast.java [hunk 10]
```diff
@@ -29,8 +30,8 @@ protected void initialize() {
     }
 
     protected void execute() {
-        translate = OI.joystick.getRawAxis(1);
-        rotate = OI.joystick.getRawAxis(4);
+        translate = OI.joystick.getRawAxis(PortNumbers.JOYSTICK_TRANSLATE_AXIS);
+        rotate = OI.joystick.getRawAxis(PortNumbers.JOYSTICK_ROTATE_AXIS);
```
Comment (simon-andrews) [inline src/org/robockets/steamworks/commands/GottaGoFast.java]: Make names more Xbox-y.

**Review by simon-andrews — CHANGES_REQUESTED** [review]

**Review by simon-andrews — APPROVED** [review]

**Comment by bshin100** [comment]
👍

**Pull request merged**